docs/topics/collection-grouping.md

```diff
@@ -30,21 +41,35 @@ right before the operation execution.
 
 Namely, `Grouping` supports the following operations:
 
-* [`eachCount()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/each-count.html) counts the elements in each group. 
+* [`eachCount()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/each-count.html) counts the elements in each group.
 * [`fold()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/fold.html) and [`reduce()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/reduce.html)
-   perform [fold and reduce](collection-aggregate.md#fold-and-reduce) operations on each group as a separate collection
-   and return the results.
+  perform [fold and reduce](collection-aggregate.md#fold-and-reduce) operations on each group as a separate collection
+  and return the results.
 * [`aggregate()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/aggregate.html) applies a given operation
-   subsequently to all the elements in each group and returns the result.
-   This is the generic way to perform any operations on a `Grouping`. Use it to implement custom operations when fold or reduce are not enough.
+  subsequently to all the elements in each group and returns the result.
+  This is the generic way to perform any operations on a `Grouping`. Use it to implement custom operations when fold or reduce are not enough.
 
-```kotlin
+You can use the `for` operator on the resulting `Map` to iterate through the groups created by the `groupingBy()` function.
+This allows you to access each key and the count of elements associated with that key.
+
+The following example demonstrates how to group strings by their first letter using the `groupingBy()` function,
+count the elements in each group, and then iterate through each group to print the key and count of elements:
 
+```kotlin
 fun main() {
 //sampleStart
-    val numbers = listOf("one", "two", "three", "four", "five", "six")
-    println(numbers.groupingBy { it.first() }.eachCount())
+    val numbers = listOf("one", "two", "three", "four", "five")
+
+    // Groups the strings by their first letter using groupingBy() and counts the elements in each group
+    val grouped = numbers.groupingBy { it.first() }.eachCount()
+
+    // Iterates through each group and prints the key and its associated values
+    for ((key, count) in grouped) {
+        println("Key: $key, Count: $count")
```

Comment:
Same as before, we can add the actual output to the code snippet. You can see some examples here: https://kotlinlang.org/docs/kotlin-tour-hello-world.html#string-templates